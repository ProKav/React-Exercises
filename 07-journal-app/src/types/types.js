export const types = {
	login: '[Auth] Login',
	logout: '[Auth] Logout',

	uiSetError: '[UI] Set Error',
	uiRemoveError: '[UI] Remove Error',

	uiStartLoading: '[UI] Start Loading',
	uiFinishLoading: '[UI] Finish Loading',

	notesAddNew: '[Notes] New Note',
	notesActive: '[Notes] Active Note',
	notesLoad: '[Notes] Load Note',
	notesUpdated: '[Notes] Updated Note',
	notesFileUrl: '[Notes] image url',
	notesDelete: '[Notes] Delete Note',
	notesLogoutCleaning: '[Notes] Cleaning Note',
}
