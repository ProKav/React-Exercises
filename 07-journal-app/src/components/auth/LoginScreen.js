import React from 'react'
import {useDispatch, useSelector} from 'react-redux'
import { Link } from 'react-router-dom'
import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth'
import { useForm } from '../../hooks/useForm'
import {removeError, setError} from "../../actions/ui";
import validator from 'validator'

export const LoginScreen = () => {
	const dispatch = useDispatch()
	const {msgError} = useSelector( state => state.ui )

	const [formValues, handleInputChange] = useForm({
		email: 'kamilo923@hotmail.com',
		password: '123456',
	})

	const { email, password } = formValues

	const handleLogin = e => {
		e.preventDefault()

		if(isFormValid()){
			dispatch(startLoginEmailPassword(email, password))
		}
	}

	const handleGoogleLogin = () => {
		dispatch(startGoogleLogin())
	}

	const isFormValid = () => {
		if(email.trim().length === 0){
			dispatch(setError('Email is required'))
			return false
		} else if(!validator.isEmail(email)){
			dispatch(setError('Email is not valid'))
			return false
		}

		dispatch(removeError())
		return true
	}

	return (
		<>
			<h3 className='auth__title'>Login</h3>

			<form onSubmit={handleLogin}>

				{
					msgError &&
					(
						<div className='auth__alert-error'>
							{msgError}
						</div>
					)
				}

				<input
					className='auth__input'
					type='text'
					placeholder='Email'
					name='email'
					autoComplete='off'
					value={email}
					onChange={handleInputChange}
				/>
				<input
					className='auth__input'
					type='password'
					placeholder='Password'
					name='password'
					autoComplete='off'
					value={password}
					onChange={handleInputChange}
				/>
				<button type='submit' className='btn btn-primary btn-block'>
					Login
				</button>

				<div className='auth__social-networks'>
					<p>Login With Social Networks</p>
					<div className='google-btn' onClick={handleGoogleLogin}>
						<div className='google-icon-wrapper'>
							<img
								className='google-icon'
								src='https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg'
								alt='google button'
							/>
						</div>
						<p className='btn-text'>
							<b>Sign in with google</b>
						</p>
					</div>
					<Link to='/auth/register' className='link'>
						Create new account
					</Link>
				</div>
			</form>
		</>
	)
}
