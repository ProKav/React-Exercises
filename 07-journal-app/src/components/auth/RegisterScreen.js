import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'
import { useDispatch, useSelector } from 'react-redux'
import validator from 'validator'
import { removeError, setError } from '../../actions/ui'
import { startRegisterWithEmailPasswordName } from '../../actions/auth'

export const RegisterScreen = () => {
	const dispatch = useDispatch()
	const { msgError } = useSelector(state => state.ui)

	const [formValues, handleInputChange] = useForm({
		name: '',
		email: '',
		password: '',
		password2: '',
	})

	const { name, email, password, password2 } = formValues

	const handleRegister = e => {
		e.preventDefault()

		if (isFormValid()) {
			dispatch(startRegisterWithEmailPasswordName(email, password, name))
		}
	}

	const isFormValid = () => {
		if (name.trim().length === 0) {
			dispatch(setError('Name is required'))
			return false
		} else if (!validator.isEmail(email)) {
			dispatch(setError('Email is not valid'))
			return false
		} else if (password.length <= 5) {
			dispatch(setError('Password should be at least 6 characters'))
			return false
		} else if (password !== password2) {
			dispatch(setError('Passwords should be equals'))
			return false
		}

		dispatch(removeError())
		return true
	}

	return (
		<div>
			<h3 className='auth__title'>Register</h3>

			<form
				onSubmit={handleRegister}
				className='animate__animated animate__fadeIn animate__slow'>
				{msgError && (
					<div className='auth__alert-error'>{msgError}</div>
				)}
				<input
					className='auth__input'
					type='text'
					placeholder='Name'
					name='name'
					autoComplete='off'
					value={name}
					onChange={handleInputChange}
				/>
				<input
					className='auth__input'
					type='text'
					placeholder='Email'
					name='email'
					autoComplete='off'
					value={email}
					onChange={handleInputChange}
				/>
				<input
					className='auth__input'
					type='password'
					placeholder='Password'
					name='password'
					autoComplete='off'
					value={password}
					onChange={handleInputChange}
				/>
				<input
					className='auth__input'
					type='password'
					placeholder='Confirm Password'
					name='password2'
					autoComplete='off'
					value={password2}
					onChange={handleInputChange}
				/>
				<button
					type='submit'
					className='btn btn-primary btn-block mt-5 mb-1'>
					Login
				</button>
				<Link to='/auth/login' className='link'>
					Already Registered?
				</Link>
			</form>
		</div>
	)
}
