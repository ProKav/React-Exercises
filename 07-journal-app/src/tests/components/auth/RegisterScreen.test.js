import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { mount } from 'enzyme'
import { RegisterScreen } from '../../../components/auth/RegisterScreen'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import '@testing-library/jest-dom'
import { types } from '../../../types/types'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {
	auth: {},
	ui: {
		loading: false,
		msgError: null,
	},
}

let store = mockStore(initState)

const wrapper = mount(
	<Provider store={store}>
		<MemoryRouter>
			<RegisterScreen />
		</MemoryRouter>
	</Provider>,
)

describe('Test in <RegisterScreen /> component', () => {
	test('Sould display <RegisterScreen /> correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})

	test('Should dispatch respect action', () => {
		const emailField = wrapper.find('input[name="email"]')

		emailField.simulate('change', {
			target: {
				value: '',
				name: 'email',
			},
		})

		wrapper.find('form').simulate('submit', {
			preventDefault() {},
		})

		const actions = store.getActions()

		expect(actions[0]).toEqual({
			type: types.uiSetError,
			payload: 'Name is required',
		})
	})
})
