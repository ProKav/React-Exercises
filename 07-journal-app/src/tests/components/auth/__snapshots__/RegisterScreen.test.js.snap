// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RegisterScreen /> component Sould display <RegisterScreen /> correctly 1`] = `
<div>
  <h3
    className="auth__title"
  >
    Register
  </h3>
  <form
    className="animate__animated animate__fadeIn animate__slow"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block mt-5 mb-1"
      type="submit"
    >
      Login
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already Registered?
    </a>
  </form>
</div>
`;
